### PR TITLE
v0.18.0

### PR DESCRIPTION
**Pull request by majorgreys**
This PR merges `0.18-dev` into `master`

### CODE REVIEW

**Comment by brettlangdon** [comment]
This lgtm, just waiting until we can release to approve

**Comment by Kyle-Verhoog** [comment]
How are my merges still not being squashed.. I keep having to switch the kind of merge button over 😕.

**Comment by brettlangdon** [comment]
@kyle yeah... I was going to merge shame you... you really just want to get those commit numbers up :p

File: ddtrace/contrib/molten/wrappers.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+class WrapperComponent(wrapt.ObjectProxy):
+    """ Tracing of components """
+    def can_handle_parameter(self, *args, **kwargs):
+        func = self.__wrapped__.can_handle_parameter
+        cname = func_name(self.__wrapped__)
+        resource = '{}.{}'.format(cname, func.__name__)
+        return trace_wrapped(resource, func, *args, **kwargs)
+
+    # TODO[tahir]: the signature of a wrapped resolve method causes DIError to
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/molten/wrappers.py]: Is this an outdated comment?

File: tests/contrib/molten/test_molten.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        self.assertIsNone(span.get_tag(errors.ERROR_MSG))
+        self.assertEqual(route_error_span.get_tag(errors.ERROR_MSG), 'Error message')
+
+    def test_resources(self):
+        """ Tests request has expected span resources """
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+
+        # `can_handle_parameter` appears twice since two parameters are in request
+        # TODO[tahir]: missing ``resolve` method for components
```
Comment (Kyle-Verhoog) [inline tests/contrib/molten/test_molten.py]: outdated comment?

Comment (majorgreys) [reply tests/contrib/molten/test_molten.py]: Maybe `TODO` is stronger than I intended it. We want to revisit this limitation of the current instrumentation and we have it on our backlog.

**Pull request merged**